code/forms/BasisSpamProtectionForm.php

```diff
@@ -0,0 +1,57 @@
+<?php
```

Comment:
I'm still not sure on the duplicate code in a separate form instance, only because core doesn't support an `updateForm` hook. I can't see many actually using this class, as they're likely to just use `Form` where needed. Have you submitted something against core to see what they say?